src/azure-cli-core/azure/cli/core/azclierror.py

```diff
@@ -179,6 +179,29 @@ class ClientRequestError(UserFault):
     pass
 
 
+# OS error types
+class AzFileNotFoundError(UserFault):
```

Comment:
Just call out for more discussions here.

If we need a same error with the system exception, what should be the naming rules.
In this case,  we just use the `FileNotFoundError` or a new name?

@yungezz @fengzhou-msft @jiasli  what's your ideas?

Reply:
Thanks for fixing the error. The `Az` prefix is used to distinguish with something? I prefer type string without `Az` since all errors are in scope of Azure CLI.

Reply:
Very strange name to have `Az` prefix.

Reply:
@houk-ms Do we need to redefine all Python built-in exceptions such as these [OS exceptions](https://docs.python.org/3/library/exceptions.html#os-exceptions)? Is it possible to use the original exception type while classifying them as UserFault in exception handler?

Reply:
If we use `FileNotFoundError` as the error name, this will shadow built-in name. For some existing code using the built-in exception, if somebody import our `FileNotFoundError` inadvertently, then the existing code may be confused.

BTW, I definitely agree with @fengzhou-msft. There exist large amounts of built-in exceptions which will be a heavy duplication if we redefine them all. Can we just wrap the original exception around `UserFault`?

Reply:
All right, make sense to me.

So let me clarify it in the developer guideline. All what we can raise in command groups are the error types
- either the[ Python Built-In error types](https://docs.python.org/3/library/exceptions.html)
- or [CLI define error types](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py).